src/main/java/io/reactivex/Observable.java

```diff
@@ -5923,18 +5949,20 @@ public final void blockingSubscribe(Observer<? super T> subscriber) {
     /**
      * Returns an Observable that emits buffers of items it collects from the source ObservableSource. The resulting
      * ObservableSource emits connected, non-overlapping buffers. It emits the current buffer and replaces it with a
-     * new buffer whenever the ObservableSource produced by the specified {@code closingIndicator} emits an item.
+     * new buffer whenever the ObservableSource produced by the specified {@code boundarySupplier} emits an item.
      * <p>
      * <img width="640" height="395" src="https://raw.github.com/wiki/ReactiveX/RxJava/images/rx-operators/buffer1.png" alt="">
      * <dl>
+     * If either the source ObservableSource or the boundary ObservableSource issues an onError notification the event
+     * is passed on immediately without first emitting the buffer it is in the process of assembling.
      *  <dt><b>Scheduler:</b></dt>
      *  <dd>This version of {@code buffer} does not operate by default on a particular {@link Scheduler}.</dd>
      * </dl>
      *
      * @param <B> the value type of the boundary-providing ObservableSource
      * @param boundarySupplier
      *            a {@link Callable} that produces an ObservableSource that governs the boundary between buffers.
-     *            Whenever the source {@code ObservableSource} emits an item, {@code buffer} emits the current buffer and
+     *            Whenever the supplied {@code ObservableSource} emits an item, {@code buffer} emits the current buffer and
```

Comment:
Everywhere else "source" is used to indicate the observable in `source.buffer(...)` - here we provide a `boundarySupplier` so this comment is referring to the _supplied_ observable.